Ce dataset contient les prix des maisons de la ville de Monroe Twp dans le New Jersey aux USA. Ici le prix des maisons dépend de la superficie (pieds carrés), des chambres à coucher et de l’âge de la maison (en années). 

Problème : Prédire les prix des maisons neuves en fonction de la superficie, des chambres et de l’âge.

Ici, la superficie, les chambres et l'âge sont appelés les variables indépendantes ou explicatives alors que le prix est la variable dépendante ou expliquée.

In [2]:
# Importons les librairies nécessaires
import pandas as pd
import numpy as np 
from sklearn import linear_model

In [3]:
# Chargeons le dataset
data = pd.read_csv('homeprices.csv')

In [4]:
# Jettons un coup d'oeil aux données
data

area  bedrooms  age   price
0  2600       3.0   20  550000
1  3000       4.0   15  565000
2  3200       NaN   18  610000
3  3600       3.0   30  595000
4  4000       5.0    8  760000
5  4100       6.0    8  810000

On remarque qu'il y'a une donnée manquante (NaN). Procédons donc à sa gestion.

### Remplacement de la valeur manquante par la médiane de la colonne

In [5]:
# Déterminons la colonne
data.bedrooms.median()

4.0

In [6]:
# Insérons la médiane dans la colonne
data.bedrooms = data.bedrooms.fillna(data.bedrooms.median())
data

area  bedrooms  age   price
0  2600       3.0   20  550000
1  3000       4.0   15  565000
2  3200       4.0   18  610000
3  3600       3.0   30  595000
4  4000       5.0    8  760000
5  4100       6.0    8  810000

### Construisons le modèle

In [7]:
reg = linear_model.LinearRegression()
reg.fit(data.drop('price', axis = 'columns'), data.price)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

### Trouvons le prix de la maison avec 3500 m² de surface, 4 chambres à coucher, 5 ans

In [8]:
reg.predict([[3500, 4, 5]])

array([690938.47943147])

Une maison de 3 500 m2 de surface possédant 4 chambres à coucher et ayant 5 ans d'âge coûtera 690938.47943147 selon le modèle.